Zotero/Scenes/Detail/PDF/Views/PDFReaderViewController.swift

```diff
@@ -91,19 +91,49 @@ class PDFReaderViewController: UIViewController {
     weak var coordinatorDelegate: (PdfReaderCoordinatorDelegate & PdfAnnotationsCoordinatorDelegate)?
 
     private lazy var shareButton: UIBarButtonItem = {
+        var menuChildren: [UIMenuElement] = []
+
         let share = UIBarButtonItem(image: UIImage(systemName: "square.and.arrow.up"), style: .plain, target: nil, action: nil)
-        share.isEnabled = !self.viewModel.state.document.isLocked
-        share.accessibilityLabel = L10n.Accessibility.Pdf.export
-        share.title = L10n.Accessibility.Pdf.export
+        share.accessibilityLabel = L10n.Accessibility.Pdf.share
+        share.title = L10n.Accessibility.Pdf.share
         share.tag = NavigationBarButton.share.rawValue
-        share.rx.tap
-             .subscribe(onNext: { [weak self, weak share] _ in
-                 guard let self, let share else { return }
-                 self.coordinatorDelegate?.showPdfExportSettings(sender: share, userInterfaceStyle: self.viewModel.state.interfaceStyle) { [weak self] settings in
-                     self?.viewModel.process(action: .export(settings))
-                 }
-             })
-             .disposed(by: self.disposeBag)
+
+        let deferredMenu = UIDeferredMenuElement.uncached { [weak self] elementProvider in
+            var elements: [UIMenuElement] = []
+            defer {
+                elementProvider(elements)
+            }
+            guard let self else { return }
+
+            if let parentKey = viewModel.state.parentKey {
+                let copyCitationAction = UIAction(title: L10n.Citation.copyCitation, image: .init(systemName: "doc.on.doc")) { [weak self] _ in
+                    guard let self, let coordinatorDelegate else { return }
```

Comment:
Hmm isn't the closure missing `[weak coordinatorDelegate]`? Or does this guard translate to `guard let self = self, let coordinatorDelegate = self.coordinatorDelegate else { return }` (that's why I hate these shortened versions)?

Same with `viewModel` below. Shouldn't be either be `[weak viewModel]` or `self.viewModel`?

Reply:
It's exactly what you describe from this implemented swift evolution [proposal](https://github.com/apple/swift-evolution/blob/main/proposals/0365-implicit-self-weak-capture.md)

Reply:
Right, I assumed that was the case, just forgot this was already in, since I mostly used explicit `self`s before. Alright then we can keep this like this, I don't like it, but it's the way the language is evolving, so no need to fight it.